File: frontend/src/types/portfolio.ts
export interface WatchlistItem {
  ticker: string
  price: number
  change: string
  suggestion: string
}

export interface HoldingItem {
  ticker: string
  avg_price: number
  current_price: number
  profit_pct: number
}

export interface PortfolioOverview {
  watchlist: WatchlistItem[]
  holdings: HoldingItem[]
}

File: backend/app/schemas/portfolio.py
from pydantic import BaseModel


class WatchlistItem(BaseModel):
    ticker: str
    price: float
    change: str
    suggestion: str


class HoldingItem(BaseModel):
    ticker: str
    avg_price: float
    current_price: float
    profit_pct: float


class PortfolioOverviewResponse(BaseModel):
    watchlist: list[WatchlistItem]
    holdings: list[HoldingItem]
